### PR TITLE
add namespace for UserOrderableServiceVo.

### LINKED ISSUE #2176: add namespace for UserOrderableServiceVo
releated #2117 

1.change register's namespace to take directly from OCL.
2.check if the namespace in the OCL and namespace in the user details are same
3.add and set namespace value for catalog service api 
4.for the namespace validation, use currentUserCanManageNamespace method

### PR DESCRIPTION
**Pull request by Alice1319**
fixes #2176 

1.register with same namespace:
image

get namespace from UserOrderableServiceVo same with template:
Snipaste_2024-12-10_11-06-35

2.register with diff namespace:
image

image

3.register when user's namespace is empty
image

image

### CODE REVIEW

**Comment by dengyali0125** [comment]
LGTM

**Comment by luoping-12345** [comment]
LGTM

**Comment by profoundwu** [comment]
LGTM

**Comment by WangLiNaruto** [comment]
LGTM

File: modules/api/src/main/java/org/eclipse/xpanse/api/config/ServiceTemplateEntityConverter.java [hunk 1]
```diff
@@ -82,6 +82,8 @@ public static UserOrderableServiceVo convertToUserOrderableServiceVo(
             BeanUtils.copyProperties(serviceTemplateEntity, userOrderableServiceVo);
             userOrderableServiceVo.setServiceTemplateId(serviceTemplateEntity.getId());
             userOrderableServiceVo.setIcon(serviceTemplateEntity.getOcl().getIcon());
+            userOrderableServiceVo.setNamespace(
+                    serviceTemplateEntity.getOcl().getNamespace());
```
Comment (baixinsui) [inline modules/api/src/main/java/org/eclipse/xpanse/api/config/ServiceTemplateEntityConverter.java:86]: This value should be serviceTemplateEntity.getNamespace() not the namespace in Ocl

**Comment by wuzhuanhong** [comment]
LGTM

**Comment by yuanxiaojun987** [comment]
LGTM

File: modules/api/src/main/java/org/eclipse/xpanse/api/config/ServiceTemplateEntityConverter.java (hunk 1 above)
Comment (Alice1319) [reply modules/api/src/main/java/org/eclipse/xpanse/api/config/ServiceTemplateEntityConverter.java]: tested and removed.

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**